packages/wallet/src/redux/protocols/existing-ledger-funding/reducer.ts

```diff
@@ -29,70 +38,33 @@ export const initialize = (
   ledgerId: string,
   sharedData: SharedData,
 ): ProtocolStateWithSharedData<states.ExistingLedgerFundingState> => {
-  const ledgerChannel = selectors.getChannelState(sharedData, ledgerId);
-  const theirCommitment = getLastCommitment(ledgerChannel);
-
-  const {
-    allocation: proposedAllocation,
-    destination: proposedDestination,
-  } = helpers.getLatestCommitment(channelId, sharedData);
-
-  if (ledgerChannelNeedsTopUp(theirCommitment, proposedAllocation, proposedDestination)) {
-    const { protocolState: ledgerTopUpState, sharedData: newSharedData } = initializeLedgerTopUp(
-      processId,
-      channelId,
-      ledgerId,
-      proposedAllocation,
-      proposedDestination,
-      theirCommitment.allocation,
-      sharedData,
-    );
-    return {
-      protocolState: states.waitForLedgerTopUp({
-        ledgerTopUpState,
-        processId,
-        channelId,
-        ledgerId,
-      }),
-      sharedData: newSharedData,
-    };
-  }
-
-  if (helpers.isFirstPlayer(ledgerId, sharedData)) {
-    const appFunding = craftAppFunding(sharedData, channelId);
-    const ourCommitment = proposeNewConsensus(
-      theirCommitment,
-      appFunding.proposedAllocation,
-      appFunding.proposedDestination,
-    );
-    const signResult = signAndStore(sharedData, ourCommitment);
-    if (!signResult.isSuccess) {
-      return {
-        protocolState: states.failure({ reason: 'ReceivedInvalidCommitment' }),
-        sharedData,
-      };
-    }
-    sharedData = signResult.store;
-
-    const messageRelay = sendCommitmentReceived(
-      theirAddress(ledgerChannel),
-      processId,
-      signResult.signedCommitment.commitment,
-      signResult.signedCommitment.signature,
-      EXISTING_LEDGER_FUNDING_PROTOCOL_LOCATOR,
-    );
-    sharedData = queueMessage(sharedData, messageRelay);
+  sharedData = registerChannelToMonitor(sharedData, processId, ledgerId);
+  sharedData = queueLockRequest(sharedData, lockChannelRequest({ channelId: ledgerId, processId }));
```

Comment:
Protocols can now request a lock on a channel. Currently they will wait until a lock is acquired but we can change the behavior so that a protocol gets a `CHANNEL_BUSY` action and we can handle that case appropriately.